src/librustc_mir/const_eval.rs

```diff
@@ -225,10 +229,12 @@ pub struct CompileTimeInterpreter<'mir, 'tcx> {
 }
 
 impl<'mir, 'tcx> CompileTimeInterpreter<'mir, 'tcx> {
-    fn new() -> Self {
+    fn new(const_limit: usize) -> Self {
+        let loop_detector_enabled = if const_limit == 0 { false } else { true };
         CompileTimeInterpreter {
+            loop_detector_enabled,
             loop_detector: Default::default(),
-            steps_since_detector_enabled: -STEPS_UNTIL_DETECTOR_ENABLED,
+            steps_since_detector_enabled: -const_limit.try_into().unwrap_or(isize::max_value()),
```

Comment:
should we error somewhere when the value given by the user is too large?

Reply:
A warning would be nice, but in this scope I can't emit a warning.
I am not sure if it is okay to emit a warning on both places where the constructor is called, this feels clumsy.

Reply:
Maybe this could be done at the attribute parsing site instead of here (the code here could then `unwrap()` the `try_into` conversion

Reply:
Then this will be handled with this issue #67265 wouldn't it?

Reply:
right